packager/packager.cc

```diff
@@ -773,13 +929,26 @@ class ImportTreeBuilder final {
             }
             node = child.get();
         }
-        node->source = {importedPackage.name.mangledName, loc, importType};
+
+        // If the node already has an import source, this is a conflicting import.
+        // See test/cli/package-import-conflicts/ for an example.
+        if (node->source.exists() && importType != ImportType::Self) {
+            addImportConflictError(ctx, loc, node->source.importLoc, exportFqn.parts);
+        }
+        node->source = {importedPackage.name.mangledName, loc, importType, isEnumeratedImport};
     }
 
+    // Method that makes a wrapper module for an import, of the form:
+    // module PkgMangledName::PkgName
+    //   Import = ImportedPkgMangledName::ImportedPkgName::Export
+    // end
+    //
+    // This method in fact creates a wrapper module recursively for the entire import tree of a package.
     void makeModule(core::Context ctx, ImportTree *node, vector<core::NameRef> &parts, ast::ClassDef::RHS_store &modRhs,
-                    ImportType moduleType, ImportTree::Source parentSrc) {
+                    ModuleType moduleType, ImportTree::Source parentSrc) {
         auto newParentSrc = parentSrc;
-        if (node->source.exists() && !parentSrc.exists()) {
+        ImportTree::Source &source = node->source;
```

Comment:
Is naming `node->source` necessary? It's adding a lot to the diff in this function. I'd also use `auto &` when initializing it.
```suggestion
        auto &source = node->source;
```

Reply:
I felt it made the code more readable. Will change to `auto`.